### PR TITLE
upgrade mysql:mysql-connector-java to 8.2.0

### PR DESCRIPTION
**Pull request by AlbericByte**
Fixes #13389.

### Description

upgrade the mysql:mysql-connector-java to 8.0.28

#### Release note
Upgrade mysql:mysql-connector-java to 8.0.28
<hr>

##### Key changed/added classes in this PR
 * `MySQLConnector.java`
 * `MySQLConnectorTest.java`
 * `pom.xml`

<hr>
This PR has:

- [x] been self-reviewed.
- [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by cryptoe** [comment]
Similar work happening in this PR as well: https://github.com/apache/druid/pull/15408/files

File: extensions-core/lookups-cached-global/src/test/java/org/apache/druid/query/lookup/namespace/JdbcExtractionNamespaceUrlCheckTest.java [hunk 1]
```diff
@@ -156,46 +156,6 @@ public boolean isEnforceAllowedProperties()
           }
       );
     }
-
-    @Test
-    public void testWhenInvalidUrlFormat()
-    {
```
Comment (FrankChen021) [inline extensions-core/lookups-cached-global/src/test/java/org/apache/druid/query/lookup/namespace/JdbcExtractionNamespaceUrlCheckTest.java:162]: why is this removed

File: extensions-core/mysql-metadata-storage/src/test/java/org/apache/druid/metadata/storage/mysql/MySQLConnectorTest.java [hunk 2]
```diff
@@ -57,16 +55,14 @@ public void testIsExceptionTransientMySql()
...
-    Assert.assertTrue(connector.connectorIsTransientException(new MySQLTransientException()));
-    Assert.assertTrue(connector.connectorIsTransientException(new MySQLTransactionRollbackException()));
     Assert.assertTrue(
         connector.connectorIsTransientException(new SQLException("some transient failure", "s0", 1317))
     );
     Assert.assertFalse(
         connector.connectorIsTransientException(new SQLException("totally realistic test data", "s0", 1337))
     );
     // this method does not specially handle normal transient exceptions either, since it is not vendor specific
-    Assert.assertFalse(
```
Comment (FrankChen021) [inline extensions-core/mysql-metadata-storage/src/test/java/org/apache/druid/metadata/storage/mysql/MySQLConnectorTest.java:69]: why is it changed from assertFalse to assertTrue

File: extensions-core/mysql-metadata-storage/src/test/java/org/apache/druid/firehose/sql/MySQLFirehoseDatabaseConnectorTest.java [hunk 3]
```diff
@@ -338,29 +337,6 @@ public boolean isEnforceAllowedProperties()
     );
   }
 
-  @Test
-  public void testFindPropertyKeysFromInvalidConnectUrl()
```
Comment (FrankChen021) [inline extensions-core/mysql-metadata-storage/src/test/java/org/apache/druid/firehose/sql/MySQLFirehoseDatabaseConnectorTest.java:342]: it makes sense to remove this duplicate test

File: extensions-core/mysql-metadata-storage/src/test/java/org/apache/druid/metadata/storage/mysql/MySQLConnectorTest.java (hunk 2 above)
Comment (AlbericByte) [reply extensions-core/mysql-metadata-storage/src/test/java/org/apache/druid/metadata/storage/mysql/MySQLConnectorTest.java:69]: Thanks for your time first.
+ We changed  MYSQL_TRANSIENT_EXCEPTION_CLASS_NAME from  "com.mysql.jdbc.exceptions.MySQLTransientException" to  "java.sql.SQLTransientException"

+ And in mysql-connector-j-8.2.0.jar, we do not have MySQLTransientException.

**Review by FrankChen021 — APPROVED** [review]
LGTM. Thanks for the contribution.
